### ISSUE TITLE
why i have not recv an alert email

### ISSUE THREAD

**Issue by wenxiaoyu** [bosun-monitor/bosun#1217]
2015/07/29 20:37:48 check.....
2015/07/29 20:37:48 check alert os.high.cpu start <nil>
2015/07/29 20:37:48 <nil>
2015/07/29 20:37:48 check alert os.high.cpu done (9.589325ms): 1 crits, 0 warns, 0 unevaluated, 0 unknown
2015/07/29 20:37:48 check.....
2015/07/29 20:37:48 check alert cpu.lookup start <nil>
2015/07/29 20:37:48 <nil>
2015/07/29 20:37:48 check alert cpu.lookup done (1.413991ms): 2 crits, 0 warns, 0 unevaluated, 0 unknown
2015/07/29 20:37:48 check took 11.199747ms
2015/07/29 20:37:53 starting check

but,i did not got the alter emal;

why?

and i test the rule,email sended.

**Comment by wenxiaoyu** [comment]
tsdbHost = 127.0.0.1:4242
graphiteHost = 
smtpHost = smtp.exmail.qq.com:25
smtpUsername=fish@warthog.cn
smtpPassword=zxc123wenxy
emailFrom = fish@warthog.cn
httpListen = :8070
timeAndDate = 202,75,179,136
stateFile = ../bosun.state
checkFrequency = 5s

notification email {
email=157085863@qq.com
next=email
timeout=1h
}
template generic {
	body = `<a href="{{.Ack}}">Acknowledge alert</a>
	<p>Alert definition:
	<p>Name: {{.Alert.Name}}
	<p>Crit: {{.Alert.Crit}}
	
	<p>Tags
	
	<table>
		{{range $k, $v := .Group}}
			{{if eq $k "host"}}
				<tr><td>{{$k}}</td><td><a href="{{$.HostView $v}}">{{$v}}</a></td></tr>
			{{else}}
				<tr><td>{{$k}}</td><td>{{$v}}</td></tr>
			{{end}}
		{{end}}
	</table>

	<p>Computation
	
	<table>
		{{range .Computations}}
			<tr><td>{{.Text}}</td><td>{{.Value}}</td></tr>
		{{end}}
	</table>`
	subject = {{.Last.Status}}: {{.Alert.Name}}: {{.Eval .Alert.Vars.q}} on {{.Group.host}}
}

alert os.high.cpu {
	template = generic
	$q = avg(q("avg:rate{counter,,1}:os.cpu{host=*}", "12h", ""))
	warn = $q > 3
	crit = $q >= 5
	critNotification = email
	warnNotification = email
}

lookup cpu {
	entry host=a,remote=b {
		high = 1
	}
	entry host=*,remote=17* {
		high = 4
	}
	entry host=matts-macbook-pro,remote=* {
		high = 2
	}
	entry host=*,remote=* {
		high = 3
	}
}

alert cpu.lookup {
	crit = lookup("cpu", "high")
}

**Comment by kylebrandt** [comment]
There should be an entry in the mail log when an email is sent, such as `Jul 29 09:33:33 ny-bosun01 bosun[24584]: 2015/07/29 09:33:33 relayed alert haproxy.server.downtime.co to [thisIsNotARealEmail@stackoverflow.com] sucessfully`

Your config looks correct to me. In quiet mode you should still get a log entry saying why the notification was not sent:

```
	if s.Conf.Quiet {
		log.Println("quiet mode prevented", len(s.notifications), "notifications")
		return
	}
```

The test page sending the email means your email server is working.... so I don't know what is going on here. Are you using our docker version? It is pretty old, I'm going to see if I can get that updated today :-/

Thanks for including all that information in your issue, a well filled bug.

**Comment by kylebrandt** [comment]
If you are using our docker version, we pushed the latest release, maybe try that and see how it goes?

**Comment by wenxiaoyu** [comment]
thank you ,i will try quit mode.
but i have not use docker.

**Comment by wenxiaoyu** [comment]
// RestoreState restores notification and alert state from the file on disk.
func (s *Schedule) RestoreState() error {
	log.Println("RestoreState")
	start := time.Now()
	s.Lock("RestoreState")
	defer s.Unlock()
	s.Search.Lock()
	defer s.Search.Unlock()
	s.Notifications = nil
	decode := func(name string, dst interface{}) error {
		log.Println("decode-->",name,dst)
		var data []byte
		err := s.db.View(func(tx *bolt.Tx) error {
 			b := tx.Bucket([]byte(dbBucket))
 			log.Println("dbBucket-->",dbBucket)
			if b == nil {
				return fmt.Errorf("unknown bucket: %v", dbBucket)
			}
			data = b.Get([]byte(name))
 			return nil
		})
		if err != nil {
			return err
		}
		gr, err := gzip.NewReader(bytes.NewReader(data))
		if err != nil {
			return err
		}
		defer gr.Close()
		log.Print("gob-->",gob.NewDecoder(gr).Decode(dst))
		return gob.NewDecoder(gr).Decode(dst)
	}

when i do this in the RestoreState method,
log.Print("gob-->",gob.NewDecoder(gr).Decode(dst))
then , print nil.....
is the reason why not send email?

**Comment by kcech** [comment]
I have similar problem with latest Bosun which I'm setting for the first time. I haven't used Docker neither. Is there some more global configuration that I'm missing? Bosun sends me mail only when I rename the alert. which gives me sensation as if there would be some long time between 2 mails can be sent.

**Comment by awesensepaul** [comment]
I have a similar issue, using the latest official `bosun` binaries (0.5.0, amd64. I had the same issue on a slightly newer `master` as well).

We are querying `elasticsearch 2.0` and sending both slack and e-mail notifications for each alert definition. In kibana when I review the `bosun` logs themselves I see (for example) `1 crits` at all the correct times (an existing service is flapping every 1 hour with an error, for example, and that is always detected as `1 crits` in the correct check interval.) However, we only receive a slack or e-mail sent for approx 5% of these alerts -- which produce a log line in the bosun logs similar to `relayed alert .. to [<EMAIL OMITTED>] successfully` (and we do receive a real email at that time)

To summarize, with the latest official release, for some of our frequent alerts we are seeing approx. 1 `relayed alert` line to every 20 `N crits` lines, where I really expected the number to be 1:1.

**Comment by awesensepaul** [comment]
Any update to this? It's a little embarrassing, not sure how long I can keep telling our team that the alert definitions are correct, but we are missing alert notifications because we are waiting for a fix in `bosun`.

**Comment by Hemantt91** [comment]
I am getting this .. notify.go:110: quiet mode prevented 1 notifications
Anyone knows how to solve it ??

**Comment by kylebrandt** [comment]
You can start bosun in quiet mode with the -q switch, and that prevents Bosun from sending notifications. We use this for:
 - Non-active bosun instances
 - Development
 - To put or production instance in quiet mode if OpenTSDB breaks (until we are able to fix).

Bosun's incident lifecycle makes it so notifications are not sent until an incident increases in severity, or is closed by a human. See http://bosun.org/usage#the-lifetime-of-an-incident
